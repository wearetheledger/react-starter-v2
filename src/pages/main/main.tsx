import { APIService } from "@api/swagger";
import { MainProps } from "@pages/main/mainContainer";
import { MainStyle } from "@pages/main/mainStyle";
import { autobind } from "core-decorators";
import * as React from "react";

interface State {}

@autobind
export class Main extends React.Component<MainProps, State> {
	public componentDidMount() {
		// Example HTTP request
		APIService.myAsset.$getMyAssetById("1").then(myAsset => {
			console.log(myAsset);
		});
	}

	public render() {
		return (
			<MainStyle>
				<h1>Home</h1>
			</MainStyle>
		);
	}
}
