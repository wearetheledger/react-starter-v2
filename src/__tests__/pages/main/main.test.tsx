import { MainContainer } from "@pages/main/mainContainer";
import React from "react";
import renderer from "react-test-renderer";

it("should match snapshot", () => {
	const wrapper = renderer.create(<MainContainer />);
	const tree = wrapper.toJSON();
	expect(tree).toMatchSnapshot();
});
