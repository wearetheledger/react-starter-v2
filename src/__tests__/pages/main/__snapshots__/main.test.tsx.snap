// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="sc-bdVaJa eAwGjB"
>
  <h1>
    Home
  </h1>
</div>
`;
